docs/src/prototypes/chatMessageWithPopover/Popover.tsx

```diff
@@ -0,0 +1,183 @@
+import {
+  Menu,
+  Popup,
+  toolbarBehavior,
+  popupFocusTrapBehavior,
+  createComponent,
+  ComponentSlotStyle,
+  ComponentVariablesInput,
+  toolbarButtonBehavior,
+} from '@stardust-ui/react'
+import { ReactChildren } from 'types/utils'
+import * as React from 'react'
+import * as cx from 'classnames'
+
+export interface PopoverProps {
+  className?: string
+}
+
+interface PopoverState {
+  focused: boolean
+  popupOpened: boolean
+}
+
+class Popover extends React.Component<PopoverProps, PopoverState> {
+  state = {
+    focused: false,
+    popupOpened: false,
+  }
+
+  changeFocusState = (isFocused: boolean) => {
+    this.state.focused !== isFocused && this.setState({ focused: isFocused })
+  }
+
+  handleFocus = () => {
+    this.changeFocusState(true)
+  }
+
+  handleBlur = e => {
+    // if e.relatedTarget === null, so the click was outside this container
```

Comment:
//if e.relatedTarget === null it means that the click was outside this container